src/routes/quickstarts/react-native/react-native-testing-library.mdx

```diff
@@ -0,0 +1,242 @@
+      server.create("user", { name: "Alice" })
+    },
+
+    routes() {
+      this.namespace = "api"
+
+      this.get("/users", schema => {
+        return schema.users.all()
+      })
+    },
+  })
+
+  return server
+}
+```
+
+> In a React Native App, put this file in `src/mirage/server.js` so that changes to it trigger rebuilds.
+
+## Step 4: Create a test file that uses Mirage
+
+Here's the component we'll be testing.
+
+```jsx
+// src/App.js
+import React, {useEffect, useState} from 'react';
+import {SafeAreaView, View, Text} from 'react-native';
+
+const App = () => {
+  let [users, setUsers] = useState([]);
+  let [serverError, setServerError] = useState();
+
+  useEffect(() => {
+    let fetchUsers = async () => {
+      try {
+        let res = await fetch('http://localhost:3000/v1/api/users');
+        let data = await res.json();
+
+        if (!res.ok) {
+          throw data;
+        }
+
+        setUsers(data.users);
+      } catch (error) {
+        setServerError(error.error);
+      }
+    };
+
+    fetchUsers();
+  }, []);
+
+  if (serverError) {
+    return (
+      <SafeAreaView style={{flex: 1}} testID="server-error">
+        <Text testID="server-error-message">{serverError}</Text>
+      </SafeAreaView>
+    );
+  }
+
+  return users.length === 0 ? (
+    <SafeAreaView style={{flex: 1}} testID="empty-users">
+      <Text>No users have been found!</Text>
+    </SafeAreaView>
+  ) : (
+    <SafeAreaView style={{flex: 1}} testID="user-list">
+      {users.map(user => {
+        return (
+          <View key={user.id}>
+            <Text>{user.name}</Text>
+          </View>
+        );
+      })}
+    </SafeAreaView>
+  );
+};
+
+export default App;
+```
+
+Create a new `src/__tests__/App.test.js` file, import your `makeServer` function, and start and shutdown Mirage using `beforeEach` and `afterEach`, making sure to pass the `test` environment to Mirage:
+
+```js
+// src/__tests__/App.test.js
+import React from "react"
+import { render, waitForElement } from "@testing-library/react"
+import App from "../App"
+import { makeServer } from "../server"
+
+let server
+
+beforeEach(() => {
+  server = makeServer({ environment: "test" })
+})
+
+afterEach(() => {
+  server.shutdown()
+})
+```
+
+## Step 5: Write tests using your Mirage server
+
+Use your tests to seed Mirage with different data scenarios, then assert against the state of your UI.
+
+> In the `test` environment, Mirage doesn't load its database `seeds`, so that the server starts out empty for each test run.
+
+```js
+it('renders the list of users from the server', async () => {
+  // Add user data to the mirage db
+  server.create('user', {name: 'Bob'});
+  server.create('user', {name: 'Alice'});
+
+  let {getByTestId} = render(<App />);
+
+  await wait(() => expect(getByTestId('empty-users')).not.toBeEmpty());
+
+  await wait(() => {
+    // Verify the number of requests handled by pretender.
+    let requests = server.pretender.handledRequests;
```

Comment:
I think we should also copy the sample tests from the other quick starts here. I don't think we should introduce `server.pretender` at this point.

Reply:
Agreed, removed.